test/functional/feature_asset_locks.py

```diff
@@ -646,8 +646,11 @@ def test_withdrawal_fork(self, node_wallet, pubkey):
         assert softfork_active(node_wallet, 'withdrawals')
 
         index = 501
-        while True:
-            self.log.info(f"Generating new Asset Unlock tx, index={index}...")
+        attempts = 0
+        while attempts < 10
+            attempts = attempts + 1
+
+            self.log.info(f"Generating new Asset Unlock tx, index={index}, attempt={attempts}...")
```

Comment:
Good catch! Could reuse `index` though e.g.
```suggestion
        while index < 511:
            self.log.info(f"Generating new Asset Unlock tx, index={index}...")
```